presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -1003,6 +1012,30 @@ public synchronized void setPartitionLeases(MetastoreContext metastoreContext, S
         throw new UnsupportedOperationException("setPartitionLeases is not supported in FileHiveMetastore");
     }
 
+    @Override
+    public long lock(MetastoreContext metastoreContext, String databaseName, String tableName)
+    {
+        HiveTableName hiveTableName = hiveTableName(databaseName, tableName);
+        while (lockedHiveTables.containsValue(hiveTableName)) {
+            try {
+                Thread.sleep(10);
+            }
+            catch (InterruptedException e) {
+                lockedHiveTables.inverse().remove(hiveTableName);
```

Comment:
hiveTableName was added by another thread, shouldn't be removed by this thread?

Reply:
Thanks for review! The removal is redundant, I'll fix it.